### ISSUE TITLE
Can the cachedkey lenght impact perfomance

### ISSUE THREAD

**Issue by delcroip** [enketo/enketo-core#982]
Dear,

for very large forms, the load time can be more that 20 seconds, on the performance monitor I see that the update function takes the most time. I thought it was reevaluating on the dependencies tree for every calculate but the line of code below suggest that there is caching

https://github.com/enketo/enketo-core/blob/425eabdc72c7040b521dc324c346ff22d85031aa/src/js/form-model.js#L1417

because the form is very large (1000+ calculate), my next best option of the performance could be the length of the cachedkey.

do you think it could be that case  ? do you think it worth spending time testing it ?

br

**Comment by lognaturel** [comment]
Can you share the form, please? We have some benchmark forms we do optimizations against (including the WHO VA form I think you know) but we know forms with other shapes could possibly be negatively impacted by optimizations. If it can’t be shared publicly, please email to me and @eyelidlessness 🙏

**Comment by delcroip** [comment]
Here is the form (the speed depends on the computer/tablet)

[adult.xlsx](https://github.com/enketo/enketo-core/files/11793126/adult.xlsx)

br

**Comment by delcroip** [comment]
If I were interested to try out improvements, Could I use an existing  test bed that is ready to measure perf / result validation?

**Comment by lognaturel** [comment]
@delcroip, what version of Enketo are you using? What device do you get a 20-30s load time on? With 6.1.0, I get a <2s load time on my laptop but I do get about 30s on an old Galaxy Tab A.

We use https://github.com/enketo/enketo-core-performance-monitor to track perf over time.

**Comment by lognaturel** [comment]
@delcroip we just released EE 6.2.0 which has additional performance improvements and that form now loads in 16s on the Galaxy Tab A. Please give it a try and share your experience. Would also be interested in what you experience @magp18

**Comment by delcroip** [comment]
Is it correct that this EE 6.2.0 is available on https://getodk.org/xlsform/ ? 

It look faster on my computer, I asked ppl with slow device to test it, I hope it is not an useless test.

A also try to take your branch of Enketo core in use it in another app but unfortunately I could not link /built it, will this enketo-core branch be publish  on npm ? 

Thank you for this work

**Comment by lognaturel** [comment]
> Is it correct that this EE 6.2.0 is available on https://getodk.org/xlsform/ ?

Yes, that's right. In fact, the performance changes released as part of 6.2.0 were already on that server when I was answering your first query even though they weren't released yet.

https://www.npmjs.com/package/enketo-core has 7.2.4 which includes everything on master. We released it last week.
